dns/py/params.py

```diff
@@ -137,6 +147,20 @@ def set(self, inputs, value):
 
     spec['args'] = args
 
+class CorednsCache(Param):
+  """
+  Changes the cache setting in the CoreDNS configmap.
+  """
+  def __init__(self):
+    super(CorednsCache, self).__init__('coredns_cache', int)
+
+  def is_relevant(self, attributes):
+    return 'cluster-dns' not in attributes
+
+  def set(self, inputs, value):
+    config = inputs.configmap_yaml['data']
+    config['Corefile'] = re.sub(r'success \d+', 'success '+`value`, config['Corefile'])
```

Comment:
Are you sure that entry inputs.configmap_yaml['data']['Corefile'] will always exist? If not then you should use.get() instead of square brackets or KeyError exception will be triggered.

Reply:
Also it is a good idea to avoid using backquotes expressions like `X`. You should better use str() or repr() because they are more readable. Also, due to obscurity they are removed in python3.